Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp

```diff
@@ -84,12 +84,18 @@ NetworkDataTaskCurl::NetworkDataTaskCurl(NetworkSession& session, NetworkDataTas
 #endif
     restrictRequestReferrerToOriginIfNeeded(request);
 
-    m_curlRequest = createCurlRequest(WTFMove(request));
-    if (!m_initialCredential.isEmpty()) {
-        m_curlRequest->setUserPass(m_initialCredential.user(), m_initialCredential.password());
-        m_curlRequest->setAuthenticationScheme(ProtectionSpace::AuthenticationScheme::HTTPBasic);
+    if (request.url().protocolIsData()) {
+        DataURLDecoder::decode(request.url(), { }, [this, protectedThis = Ref { *this }](auto decodeResult) mutable {
+            didReadDataURL(WTFMove(decodeResult));
+        });
```

Comment:
This process should execute at NetworkDataTaskCurl::resume(). Also, it should check m_state before execute didReadDataURL().